DOC-2898: Documentation enhancements: for https://www.tiny.cloud/docs/tinymce/latest/custom-toggle-toolbar-button/ (#4071)
tinymce/tinymce-docs: modules/ROOT/pages/custom-toggle-toolbar-button.adoc

= Creating custom toggle toolbar buttons
:navtitle: Toggle toolbar button
:description: Creating custom toggle toolbar buttons for {productname}
:keywords: toolbar, toolbarbuttons, buttons, toolbarbuttonsapi, toggle, formatChanged

> A toggle button triggers an action when clicked and maintains an active state. This means it can be toggled `+on+` or `+off+`. The toggle button provides the user visual feedback of its state through CSS styling. An example of this behavior is the *Bold*, which becomes highlighted when the cursor is within text that has bold formatting.  

Reviewer: Missing the word "button" after *Bold*, and there's trailing whitespace at end of line.

```suggestion
A toggle button triggers an action when clicked and maintains an active state. This means it can be toggled `+on+` or `+off+`. The toggle button provides the user visual feedback of its state through CSS styling. An example of this behavior is the *Bold* button, which becomes highlighted when the cursor is within text that has bold formatting.
```

liveDemo::custom-toolbar-toggle-button[tab="js"]

The example above adds two custom *strikethrough* toggle buttons. Both buttons use the xref:editor-command-identifiers.adoc[`+mceToggleFormat+`] command to apply and remove strikethrough formatting. This command toggles a specified format on and off, but only works for xref:content-formatting.adoc#formats[formats] already registered with the editor. In this example, `+strikethrough+` is the registered format.

> === Basic togle: manual state management

Reviewer: Typo: "togle" → "toggle"

```suggestion
=== Basic toggle: manual state management
```

The first (`+customStrikethrough+`) button applies and removes strikethrough formatting. Its state toggles upon click using `+api.setActive(!api.isActive())+`. However, this button does not reflect whether the selected content has strikethrough formatting which is its expected behavior. Moving the cursor into content with strikethrough formatting does not activate the button, and moving it out does not deactivate it.

> === State-synced toggle: automatic state updates 

Reviewer: Trailing whitespace after "updates".

```suggestion
=== State-synced toggle: automatic state updates
```

The second button (`+customToggleStrikethrough+`) addresses this by using xref:apis/tinymce.formatter.adoc#formatChanged[`+editor.formatter.formatChanged+`] in its `+onSetup+` callback to monitor the formatting state of the current selection.

> Note: The format name passed to +mceToggleFormat+ via +editor.execCommand(command, ui, args)+ is the same as the one used in +editor.formatter.formatChanged(formatName, callback)+.

Reviewer: Two issues here:
1. `Note:` should be `NOTE:` for AsciiDoc admonition rendering.
2. Inline code needs backtick wrapping (`` `+code+` ``) to match the rest of the page — bare `+code+` only does passthrough without monospace.

```suggestion
NOTE: The format name passed to `+mceToggleFormat+` via `+editor.execCommand(command, ui, args)+` is the same as the one used in `+editor.formatter.formatChanged(formatName, callback)+`.
```

The `+formatChanged+` method accepts the following parameters:

> The `+formatChanged+` method accepts the following parameters:

Reviewer: Duplicated paragraph — remove the first occurrence.

```suggestion
The `+formatChanged+` method accepts the following parameters:
```

> * `+formats+` (String, required) -- The name of the registered format to monitor.

Reviewer: The `formatChanged` first parameter accepts a comma-separated list of format names (verified against source: `Formatter.ts` and the existing API docs at `tinymce.formatter.adoc`). The current wording implies only a single name.

```suggestion
* `+formats+` (String, required) -- A comma-separated list of registered format names to monitor.
```
